subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/WellBehavedPluginTest.groovy

```diff
@@ -39,6 +39,10 @@ abstract class WellBehavedPluginTest extends AbstractIntegrationSpec {
         return "assemble"
     }
 
+    def setup() {
+        executer.noDeprecationChecks()
```

Comment:
There're several assertion in this class: `!file("build").exists()`, which is not true when running on Java 7 because deprecation report will be generated.

Reply:
Should we maybe only call this method if run on Java 7? Otherwise we might hide other deprecation messages when run run Java 8+.

Reply:
Makes sense, given that we're about to drop Java 7 support. I've changed them with `JDK8_OR_LATER` annotation.